juspay/hyperswitch
# fix(router): add routing cache invalidation on payment connector update

## Type of Change



## Description

This PR adds an invalidation for the Knowledge Graph cache by updating the merchant account when a payment connector is updated.

### Additional Changes





## Motivation and Context

If a connector is disabled, the change should show up in routing.

## How did you test it?

1. Create a Merchant Account
2. Create two merchant connector accounts X and Y
3. Make a payment with NO routing algorithm.
4. Initially the payment should go through X since it was configured first.
5. Now disable X.
6. Make another payment. This should go through Y now.

Initial Payment through X
<img width="551" alt="image" src="https://github.com/juspay/hyperswitch/assets/47862918/df770cd3-a266-4979-b522-7c8973601c2f">

Disable X
<img width="551" alt="image" src="https://github.com/juspay/hyperswitch/assets/47862918/321c2871-9119-429f-8de1-8a6aff60fc4e">

Next payment goes through Y
<img width="551" alt="image" src="https://github.com/juspay/hyperswitch/assets/47862918/83d26848-2e09-4a6b-9cf7-3bb6868d29f1">


## Checklist
